src/api/liskApi.js

```diff
@@ -12,212 +12,184 @@
-		this.nethash = this.getNethash(options.nethash);
+				: this.getDefaultPort();
+		this.randomizeNodes =
+			options.randomizeNodes !== undefined ? options.randomizeNodes : true;
+		this.providedNode = options.node || null;
+		this.node = options.node || this.selectNewNode();
+		this.headers = this.getDefaultHeaders(nethash);
+
+		// API Resource definition
+		this.accounts = new resources.AccountResource(this);
+		this.blocks = new resources.BlockResource(this);
+		this.dapps = new resources.DappResource(this);
+		this.delegates = new resources.DelegateResource(this);
+		this.signatures = new resources.SignatureResource(this);
+		this.transactions = new resources.TransactionResource(this);
+		this.voters = new resources.VoterResource(this);
+		this.votes = new resources.VoteResource(this);
 	}
 
-	getNethash(providedNethash) {
-		const { port } = this;
-		const NetHash = this.testnet
-			? utils.netHashOptions({ port }).testnet
-			: utils.netHashOptions({ port }).mainnet;
+	get allNodes() {
+		return {
+			current: this.node,
+			default: this.defaultNodes,
+			testnet: this.defaultTestnetNodes,
+		};
+	}
 
-		if (providedNethash) {
-			NetHash.nethash = providedNethash;
-			NetHash.version = '0.0.0a';
+	get currentNodes() {
+		if (this.testnet) {
+			return this.defaultTestnetNodes;
 		}
+		return this.defaultNodes;
+	}
 
-		return NetHash;
+	get urlPrefix() {
+		if (this.ssl) {
+			return 'https';
+		}
+		return 'http';
 	}
 
-	getNodes() {
-		return {
-			official: this.defaultNodes.map(node => ({ node })),
-			ssl: this.defaultSSLNodes.map(node => ({ node, ssl: true })),
-			testnet: this.defaultTestnetNodes.map(node => ({
-				node,
-				testnet: true,
-			})),
-		};
+	get fullURL() {
+		const nodeUrl = this.port ? `${this.node}:${this.port}` : this.node;
+		return `${this.urlPrefix}://${nodeUrl}`;
 	}
 
-	setNode(node) {
-		this.node = node || privateApi.selectNewNode.call(this);
-		return this.node;
+	getDefaultHeaders(providedNethash) {
+		const headers = Object.assign({}, commonHeaders, {
+			port: this.port,
+			nethash: this.testnet ? TESTNET_NETHASH : MAINNET_NETHASH,
+		});
+		if (providedNethash) {
+			headers.nethash = providedNethash;
+			headers.version = '0.0.0a';
+		}
+		return headers;
 	}
 
-	setTestnet(testnet) {
-		if (this.testnet !== testnet) {
-			this.bannedNodes = [];
+	getDefaultPort() {
+		if (this.testnet) {
+			return TEST_PORT;
+		}
+		if (this.ssl) {
+			return SSL_PORT;
+		}
+		return LIVE_PORT;
+	}
+
+	getRandomNode() {
+		const nodes = this.currentNodes.filter(node => !this.isBanned(node));
+
+		if (!nodes.length || nodes.length === 0) {
+			throw new Error(
+				'Cannot get random node: all relevant nodes have been banned.',
+			);
 		}
-		this.testnet = testnet;
-		this.port = testnet ? TEST_PORT : LIVE_PORT;
 
-		privateApi.selectNewNode.call(this);
+		const randomIndex = Math.floor(Math.random() * nodes.length);
+		return nodes[randomIndex];
+	}
+
+	setSSL(newSSLValue) {
```

Comment:
why do we need default value for setter?

Reply:
This isn't strictly a `setter`. I could imaging that people might call `.setSSL()` thinking it would default to `true`. And if they do that, `undefined` is going to trigger a new node selection whether `ssl` is currently `true` or `false`.